provider/pkg/provider/provider.go

```diff
@@ -95,19 +106,19 @@ func (k *commandProvider) StreamInvoke(req *pulumirpc.InvokeRequest, server pulu
 // the provider inputs are using for detecting and rendering diffs.
 func (k *commandProvider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (*pulumirpc.CheckResponse, error) {
 	urn := resource.URN(req.GetUrn())
-	ty := urn.Type()
-	if ty != "command:local:Command" && ty != "command:remote:Command" {
-		return nil, fmt.Errorf("unknown resource type %q", ty)
+	err := check(urn)
+	if err == nil {
+		return &pulumirpc.CheckResponse{Inputs: req.News, Failures: nil}, nil
+	} else {
+		return nil, err
 	}
-	return &pulumirpc.CheckResponse{Inputs: req.News, Failures: nil}, nil
 }
 
 // Diff checks what impacts a hypothetical update will have on the resource's properties.
 func (k *commandProvider) Diff(ctx context.Context, req *pulumirpc.DiffRequest) (*pulumirpc.DiffResponse, error) {
```

Comment:
I think we'll need to update Diff to handle the new properties - and decide on replacement semantics for `localPath` and `remotePath`.